floor/writer.go

```diff
@@ -149,6 +149,11 @@ func (m *reflectMarshaller) decodeTimestampValue(elem *parquet.SchemaElement, fi
 }
 
 func (m *reflectMarshaller) decodeValue(field interfaces.MarshalElement, value reflect.Value, schemaDef *parquetschema.SchemaDefinition) error {
+	elem := schemaDef.SchemaElement()
+	if elem == nil {
```

Comment:
this is to allow working with structs that are only partially described by a schema so you can write to/read from just the schema-defined fields without raising errors on undefined fields.